qiskit/opflow/converters/circuit_sampler.py

```diff
@@ -147,34 +152,45 @@ def convert(self,
         Raises:
             OpflowError: if extracted circuits are empty.
         """
-        if self._last_op is None or id(operator) != id(self._last_op):
-            # Clear caches
-            self._last_op = operator
-            self._reduced_op_cache = None
-            self._circuit_ops_cache = None
-            self._transpiled_circ_cache = None
-            self._transpile_before_bind = True
-
-        if not self._reduced_op_cache:
+        # check if the operator should be cached
+        op_id = hash(operator)
```

Comment:
See the comment around hash in operator base. When I first saw this and thought hash my first thought was that hash collisions can exist which makes this suspect - I saw later its not what I thought it was...

Reply:
Resolved in b585a0f